liquidprompt

```diff
@@ -3055,8 +3092,13 @@ prompt_on() {
 
         # Disable parameter/command expansion from PS1
         shopt -u promptvars
-        PROMPT_COMMAND=__lp_set_prompt
-        (( LP_DEBUG_TIME )) && PROMPT_COMMAND="time $PROMPT_COMMAND" || true
+        if ! __lp_use_bash_preexec; then
+            PROMPT_COMMAND=__lp_set_prompt
+            (( LP_DEBUG_TIME )) && PROMPT_COMMAND="time $PROMPT_COMMAND" || true
+        else
+            precmd_functions+=(__lp_set_prompt)
+            (( LP_DEBUG_TIME )) && precmd_functions+=("time __lp_set_prompt") || true
```

Comment:
You're right:
```
± type -t __lp_set_prompt
function
[ismith:~/bin/liquidprompt/tests] [sts] [⎈arn:aws:eks:us] ismith.compatibility-with-bash-preexec
± type -t "__lp_set_prompt"
function
[ismith:~/bin/liquidprompt/tests] [sts] [⎈arn:aws:eks:us] ismith.compatibility-with-bash-preexec
± type -t "time __lp_set_prompt"
[ismith:~/bin/liquidprompt/tests] [sts] [⎈arn:aws:eks:us] ismith.compatibility-with-bash-preexec 1
± 
```

Do you think this would be better done by defining:
```
__lp__timed_set_prompt() {
  time __lp_set_prompt
}
```
and using that in this case?

Reply:
3a02470